rasa/core/lock_store.py

```diff
@@ -305,7 +311,17 @@ def _create_from_endpoint_config(
 
         lock_store: LockStore = InMemoryLockStore()
     elif endpoint_config.type == "redis":
+        rasa.shared.utils.io.raise_deprecation_warning(
+            "'RedisLockStore' is deprecated and will be removed in Rasa Open "
+            "Source 4.0.0. Please migrate to the enhanced 'ConcurrentRedisLockStore'.",
+            docs=DOCS_URL_MIGRATION_GUIDE,
+        )
+
         lock_store = RedisLockStore(host=endpoint_config.url, **endpoint_config.kwargs)
+    elif endpoint_config.type == "concurrent-redis":
+        lock_store = ConcurrentRedisLockStore(
+            host=endpoint_config.url, **endpoint_config.kwargs
+        )
```

Comment:
from a UX standpoint, should we rather allow people to configure the same `redis` endpoint, but with a `concurrent: true|false` key that is popped in this method, something like this:
```yml
lock_store:
  type: "redis"
  concurrent: true
  ...other kwargs...
```

That way, when 4.0.0 comes, we ask a very minimal effort from users to migrate. With this implementation, the function should still issue a warning if `concurrent` is `False` or is not set, the warning message would be different though.

(sorry I was just passing by and I saw this!)

Reply:
Good point, I can apply this suggestion, although users would still need to do a db migration when upgrading to the new lock store.